file_handler.c

```diff
@@ -6,13 +6,17 @@
 #include <line_parser.h>
 #include <ctype.h>
 
-int is_empty_line(char *line) {
-    for (int i = 0; i < strlen(line); i++) {
-        if (!isspace(line[i])) {
-            return 0;
-        }
+/* Some of the functions are not declared anywhere. I guess it should go to the file_handler.h */
+
+int is_empty_line(char *line) { /* I changed it because fgets contains the \n and in your code I think we would have always gotten 0 because of that */

```

Comment:
the previous version works fine, I added a test

Reply:
OK